packages/auth/src/mfa/assertions/totp.ts

```diff
@@ -155,10 +158,20 @@ export class TotpMultiFactorAssertionImpl
 
   /** @internal */
   _finalizeSignIn(
```

Comment:
Please mark this and `_finalizeEnroll` as `async` and try `.to.be.rejectedWith()` again in https://github.com/firebase/firebase-js-sdk/pull/6640/

This is a really subtle mistake to make, but putting `async` in front of the function name will ensure that _every_ exception is thrown as a promise rejection, which is what we want